db_processor.bal

```diff
@@ -303,13 +323,31 @@ function addAidPackageItem(AidPackageItem aidPackageItem) returns int|error {
 }
 
 function constructAidPAckageItem(int packageId, AidPackageItem aidPackageItem) returns error? {
+    string errorMessage = "";
     Quotation quotation = check getQuotation(aidPackageItem.quotationID);
     aidPackageItem.quotation = quotation;
     aidPackageItem.packageID = packageId;
     aidPackageItem.totalAmount = <decimal>aidPackageItem.quantity * quotation.unitPrice;
     aidPackageItem.packageItemID = check addAidPackageItem(aidPackageItem);
-    check updateMedicalNeedQuantity(aidPackageItem.needID);
-    check updateQuotationRemainingQuantity(aidPackageItem);
+    transaction {
+        error? updateMedicalQuantityResult =  updateMedicalNeedQuantity(aidPackageItem.needID);
+        error? updateQuotationQuantityResult = updateQuotationRemainingQuantity(aidPackageItem);
+        if (updateMedicalQuantityResult is error) {
+            errorMessage += generateTransactionErrorMessageForUpdateOperation(updateMedicalQuantityResult);
+        }
+        if (updateQuotationQuantityResult is error) {
+            errorMessage += generateTransactionErrorMessageForUpdateOperation(updateQuotationQuantityResult);
+        }
+        if (errorMessage != "") {
+            rollback;
+            return error(errorMessage);
+        } else {
+            error? err = commit;
+            if err is error {
+                return error("Error occurred while committing");
```

Comment:
This error message is too generic. Check all the places.

Reply:
Add the error message to the end of the value. Please check and give your thoughts.